src/main/java/com/rapid7/client/dcerpc/mslsad/LocalSecurityAuthorityService.java

```diff
@@ -40,19 +44,50 @@
 public class LocalSecurityAuthorityService {
 
     public LocalSecurityAuthorityService(final RPCTransport transport) {
-        this.transport = transport;
+	this.transport = transport;
+    }
+
+    public PolicyAuditEventsInfo getAuditPolicy(ContextHandle handle) throws IOException {
+	if (handle == null) {
```

Comment:
You could consolidate these checks into something like `checkHandle(ContextHandle)`

Reply:
@tosmun-r7 , something like ? Thanks!
```
 public void checkHandle(ContextHandle handle) throws IOException {
	if (handle == null) {
            throw new IllegalArgumentException("ContextHandle is invalid: " + handle);
        }
    }
```
```
    public PolicyAuditEventsInfo getAuditPolicy(ContextHandle handle) throws IOException {
	checkHandle(handle);

	final LsarQueryInformationPolicyRequest queryRequest = new LsarQueryInformationPolicyRequest(handle,
		POLICY_AUDIT_EVENTS_INFORMATION);
	final PolicyAuditEventsInformationResponse queryResponse = transport.call(queryRequest);
	return queryResponse.getPolicyAuditInformation();
    }
```

Reply:
Exactly :+1:

Reply:
This can be private